generators/entity-client/templates/react/src/main/webapp/app/entities/entity-update.tsx.ejs

```diff
@@ -227,11 +227,17 @@ export const <%= entityReactName %>Update = (props: I<%= entityReactName %>Updat
       <Col md="8">
         { loading ? <p>Loading...</p> :
         <AvForm model={isNew ? {} : <%= entityInstance %>Entity} onSubmit={saveEntity} >
-          <%_ for (field of fields) {
+          { !isNew ?
+            <AvGroup>
+              <Label for="<%= entityFileName %>-id"><Translate contentKey="global.field.id">ID</Translate></Label>
+              <AvInput id="<%= entityFileName %>-id" type="text" className="form-control" name="id" required readOnly/>
```

Comment:
`name="id"` will break custom ids support.
Should use `${ primaryKey.name }`.

Reply:
Like this?

```jsx
<AvGroup>
  <Label for="<%= entityFileName %>-<%= primaryKey.name %>"><Translate contentKey="global.field.<%= primaryKey.name %>">ID</Translate></Label>
  <AvInput id="<%= entityFileName %>-<%= primaryKey.name %>" type="text" className="form-control" name="<%= primaryKey.name %>" required readOnly/>
</AvGroup>
```

Reply:
I've put another suggestion below.